qiskit/algorithms/time_evolvers/variational/solvers/var_qte_linear_solver.py

```diff
@@ -0,0 +1,134 @@
+                imaginary part is expected.
+
+        Raises:
+            TypeError: If t_param is provided and Hamiltonian is not of type SparsePauliOp.
+        """
+        self._var_principle = var_principle
+        self._hamiltonian = hamiltonian
+        self._ansatz = ansatz
+        self._gradient_params = gradient_params
+        self._bind_params = gradient_params
+        self._time_param = t_param
+        self.lse_solver = lse_solver
+        self._imag_part_tol = imag_part_tol
+
+        if self._time_param is not None and not isinstance(self._hamiltonian, SparsePauliOp):
+            raise TypeError(
+                f"A time parameter {t_param} has been specified, so a time-dependent "
+                f"hamiltonian was expected. The operator provided is of type {type(self._hamiltonian)}, "
+                f"which does not support parametrization. "
+                f"Please provide the parametrized hamiltonian as a SparsePauliOp."
+            )
+
+    @property
+    def lse_solver(self) -> Callable[[np.ndarray, np.ndarray], np.ndarray]:
+        """Returns an LSE solver callable."""
+        return self._lse_solver
+
+    @lse_solver.setter
+    def lse_solver(self, lse_solver: Callable[[np.ndarray, np.ndarray], np.ndarray] | None) -> None:
+        """Sets an LSE solver. Uses a ``np.linalg.lstsq`` callable if ``None`` provided."""
+        if lse_solver is None:
+            lse_solver = lambda a, b: np.linalg.lstsq(a, b, rcond=1e-2)[0]
+
+        self._lse_solver = lse_solver
+
+    def solve_lse(
+        self,
+        param_dict: dict[Parameter, float],
+        time_value: float | None = None,
+    ) -> (list | np.ndarray, list | np.ndarray, np.ndarray):
+        """
+        Solve the system of linear equations underlying McLachlan's variational principle for the
+        calculation without error bounds.
+
+        Args:
+            param_dict: Dictionary which relates parameter values to the parameters in the ansatz.
+            time_value: Time value that will be bound to ``t_param``. It is required if ``t_param``
+                is not ``None``.
+
+        Returns:
+            Solution to the LSE, A from Ax=b, b from Ax=b.
+
+        Raises:
+            ValueError: If no time value is provided for time dependent hamiltonians.
+
+        """
+        param_values = list(param_dict.values())
+        metric_tensor_lse_lhs = self._var_principle.metric_tensor(self._ansatz, param_values)
+        hamiltonian = self._hamiltonian
+
+        if self._time_param is not None:
+            if time_value is not None:
+                parametrized_coeffs = copy.deepcopy(self._hamiltonian.coeffs)
+                bound_params_array = assign_parameters(parametrized_coeffs, [time_value])
+                hamiltonian = SparsePauliOp(self._hamiltonian.paulis, bound_params_array)
+            else:
+                raise ValueError(
+                    f"Providing a time_value is required for time-dependant hamiltonians,"
```

Comment:
```suggestion
                    f"Providing a time_value is required for time-dependant hamiltonians, "
```
Trailing space for when concatenated with next lin